cmd/kubectl-testkube/commands/testworkflows/run.go

```diff
@@ -0,0 +1,185 @@
+	var (
+		executionName string
+		config        map[string]string
+		watchEnabled  bool
+		silentMode    bool
+	)
+
+	cmd := &cobra.Command{
+		Use:     "testworkflow [name]",
+		Aliases: []string{"testworkflows", "tw"},
+		Args:    cobra.ExactArgs(1),
+		Short:   "Starts test workflow execution",
+
+		Run: func(cmd *cobra.Command, args []string) {
+			namespace := cmd.Flag("namespace").Value.String()
+			client, _, err := common.GetClient(cmd)
+			ui.ExitOnError("getting client", err)
+
+			name := args[0]
+			execution, err := client.ExecuteTestWorkflow(name, testkube.TestWorkflowExecutionRequest{
+				Name:   executionName,
+				Config: config,
+			})
+			ui.ExitOnError("execute test workflow "+name+" from namespace "+namespace, err)
+			err = render.Obj(cmd, execution, os.Stdout, renderer.TestWorkflowExecutionRenderer)
+			ui.ExitOnError("render test workflow execution", err)
+
+			if watchEnabled {
+				ui.NL()
+				result, err := watchTestWorkflowLogs(execution.Name, execution.Signature, client) // TODO(TKC-1652): Use execution.Id when will be replaced
+				ui.ExitOnError("reading test workflow execution logs", err)
+
+				// Apply the result in the execution
+				execution.Result = result
+				if result.IsFinished() {
+					execution.StatusAt = result.FinishedAt
+				}
+
+				// Display message depending on the result
+				switch {
+				case result.Initialization.ErrorMessage != "":
+					ui.Warn("test workflow execution failed:\n")
+					ui.Errf(result.Initialization.ErrorMessage)
+					os.Exit(1)
+				case result.IsFailed():
+					ui.Warn("test workflow execution failed")
+					os.Exit(1)
+				case result.IsAborted():
+					ui.Warn("test workflow execution aborted")
+					os.Exit(1)
+				case result.IsPassed():
+					ui.Success("test workflow execution completed with success in " + result.FinishedAt.Sub(result.QueuedAt).String())
+				}
+			}
+		},
+	}
+
+	cmd.Flags().StringVarP(&executionName, "name", "n", "", "execution name, if empty will be autogenerated")
+	cmd.Flags().StringToStringVarP(&config, "env", "", map[string]string{}, "configuration variables in a form of name1=val1 passed to executor")
+	cmd.Flags().BoolVarP(&watchEnabled, "watch", "f", false, "watch for changes after start")
+	cmd.Flags().BoolVarP(&silentMode, "silent", "", false, "don't print intermediate test execution")
+
+	return cmd
+}
+
+func flattenSignatures(sig []testkube.TestWorkflowSignature) []testkube.TestWorkflowSignature {
+	res := make([]testkube.TestWorkflowSignature, 0)
+	for _, s := range sig {
+		if len(s.Children) == 0 {
+			res = append(res, s)
+		} else {
+			res = append(res, flattenSignatures(s.Children)...)
+		}
+	}
+	return res
+}
+
+func printResultDifference(res1 *testkube.TestWorkflowResult, res2 *testkube.TestWorkflowResult, steps []testkube.TestWorkflowSignature) bool {
+	if res1 == nil || res2 == nil {
+		return false
+	}
+	changed := false
+	for i, s := range steps {
+		r1 := res1.Steps[s.Ref]
```

Comment:
is Steps always initialized?

Reply:
It is `map[string]TestWorkflowStepResult`, so even if it's not available it will return empty object